relayer/src/supervisor.rs

```diff
@@ -278,40 +303,70 @@ impl From<UnidirectionalChannelPath> for Object {
 }
 
 impl Object {
+    pub fn src_chain_id(&self) -> &ChainId {
+        match self {
+            Self::UnidirectionalChannelPath(ref path) => &path.src_chain_id,
+        }
+    }
+
+    pub fn dst_chain_id(&self) -> &ChainId {
+        match self {
+            Self::UnidirectionalChannelPath(ref path) => &path.dst_chain_id,
+        }
+    }
+
     /// Build the object associated with the given [`SendPacket`] event.
-    pub fn for_send_packet(e: &SendPacket, chain_id: &ChainId) -> Self {
+    pub fn for_send_packet(e: &SendPacket, src_chain: &dyn ChainHandle) -> Self {
+        let dst_chain_id =
+            get_counterparty_chain(src_chain, &e.packet.source_channel, &e.packet.source_port)
+                .unwrap();
+
         UnidirectionalChannelPath {
-            src_chain_id: chain_id.clone(),
+            dst_chain_id,
+            src_chain_id: src_chain.id(),
             src_channel_id: e.packet.source_channel.clone(),
             src_port_id: e.packet.source_port.clone(),
         }
         .into()
     }
 
     /// Build the object associated with the given [`WriteAcknowledgement`] event.
-    pub fn for_write_ack(e: &WriteAcknowledgement, chain_id: &ChainId) -> Self {
+    pub fn for_write_ack(e: &WriteAcknowledgement, src_chain: &dyn ChainHandle) -> Self {
+        let dst_chain_id =
+            get_counterparty_chain(src_chain, &e.packet.source_channel, &e.packet.source_port)
```

Comment:
Testing fails with different channel ids and multiple chains. Here we should have:
```suggestion
            get_counterparty_chain(src_chain, &e.packet.destination_channel, &e.packet.destination_port)
```